### PR TITLE
Get package by filename

### PR DESCRIPTION
**Pull request by JeremyLecoeur**
# Description

Expose the newly created endpoint to get a package by a filename (needed for automatic update of SPARC metadata)

## Type of change

- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: tests/test_io.py [hunk 1]
```diff
@@ -30,6 +32,22 @@ def test_upload_legacy_to_dataset(dataset):
...
+    ([FILE4], 1),          # Single file
+])
+def test_get_by_filename(dataset, upload_args, n_files):
+    """
+    Note: ETL will fail since destination will likely be removed
+          before being processed.
+    """
+    dataset.upload(*upload_args)
+    dataset.update()
+    time.sleep(1)
```
Comment (rlmv) [inline tests/test_io.py]: Depending on any particular sleep interval will probably cause intermittent test failures down the line. Could this wait until the package is ready?

**Comment by rlmv** [comment]
Instead of returning a `PackagesReponse` object with a nested `packages` key that the unser needs to unpacke could `get_packages_by_filename` just return a list of packages instead? That would both remove an extra class from the code, and simplify interacting with this endpoint.

File: tests/test_io.py (hunk 1 above)
Comment (JeremyLecoeur) [reply tests/test_io.py]: done

File: tests/test_io.py [hunk 2]
```diff
@@ -30,6 +32,27 @@ def test_upload_legacy_to_dataset(dataset):
...
+    """
+    dataset.upload(*upload_args)
+    dataset.update()
+    for i in range(5):
+        pp = dataset.get_items_by_name("test-78f3ea50-b")
+        if (pp[0].state == 'READY'):
+            break
+        else:
+            time.sleep(0.5)
+    p = dataset.get_packages_by_filename('test-78f3ea50-b')
```
Comment (rlmv) [inline tests/test_io.py]: Interesting, the endpoint doesn't let you include the extension in the filename when making requests? Is that a bug?

File: blackfynn/api/data.py [hunk 3]
```diff
@@ -73,8 +74,12 @@ def user_collaborators(self,ds):
         resp = self._get( self._uri('/{id}/collaborators/users', id=id))
         return [UserCollaborator.from_dict(u) for u in resp]
 
-    def owner(self, ds):
+    def get_packages_by_filename(self,ds,filename):
+        id = self._get_id(ds)
+        resp = self._get( self._uri('/{id}/packages?filename={filename}', id=id, filename=filename))
```
Comment (mikeswoods) [inline blackfynn/api/data.py]: Does filename refer to the basename of the file (sans extension) or the whole thing? Can you do something like `filename=foo.txt`?

Comment (JeremyLecoeur) [reply blackfynn/api/data.py]: The underlying endpoint in bf-api only takes the basename of the file and returns all matches and lets the user deal with it

**Pull request merged**